### PR TITLE
Move `onSettingsPage` example to Preinstalled example snap

### PR DESCRIPTION
**Pull request by GuillaumeRx**
This PR removes the `settings-page-example-snap` and moves its logic to the `preinstalled-example-snap` since this handler is restricted to preinstalled snaps for now.

### CODE REVIEW

File: packages/examples/packages/preinstalled/README.md [hunk 1]
```diff
@@ -5,3 +5,26 @@ MetaMask extension by default.
...
 > Preinstalled Snaps are primarily for internal use by MetaMask.
+
+This snap also demonstrates how to use the `endowment:page-settings` permission to show a settings page that leverages custom UI components.
+
+> [!NOTE]
+> This endowment is initially restricted to preinstalled snaps only.
+
+## Snap manifest
+
+The manifest of this snap includes the `endowment:page-settings` permission:
```
Comment (FrederikBolding) [inline packages/examples/packages/preinstalled/README.md:16]: We should make a note that we likely want to have this as a separate example Snap later. E.g. once we give the capability to non preinstalled Snaps

File: packages/examples/packages/preinstalled/src/components/settings.tsx
Comment (FrederikBolding) [inline packages/examples/packages/preinstalled/src/components/settings.tsx:1]: Nit: can we capitalize the filename?

File: packages/examples/packages/preinstalled/src/index.tsx [hunk 2]
```diff
@@ -29,18 +37,62 @@ export const onRpcRequest: OnRpcRequestHandler = async ({ request }) => {
...
   }
 };
 
+/**
+ * Handle incoming settings page requests from the MetaMask clients.
+ *
+ * @returns A static panel rendered with custom UI.
+ * @see https://docs.metamask.io/snaps/reference/exports/#onsettingspage
+ */
+export const onSettingsPage: OnSettingsPageHandler = async () => {
```
Comment (FrederikBolding) [inline packages/examples/packages/preinstalled/src/index.tsx:61]: Lets add the same tests we had on the previous example

File: packages/examples/packages/preinstalled/README.md (hunk 1 above)
Comment (GuillaumeRx) [reply packages/examples/packages/preinstalled/README.md:16]: https://github.com/MetaMask/snaps/issues/2963

File: packages/examples/packages/preinstalled/src/index.tsx [hunk 3]
```diff
@@ -1,23 +1,31 @@
...
  * - `showDialog` - Opens a dialog.
+ * - `getSettings`: Get the settings state from the snap state.
  *
  * @param params - The request parameters.
  * @param params.request - The JSON-RPC request object.
  * @returns The JSON-RPC response.
  * @see https://docs.metamask.io/snaps/reference/exports/#onrpcrequest
  * @see https://docs.metamask.io/snaps/reference/rpc-api/#wallet_invokesnap
- * @see https://docs.metamask.io/snaps/reference/rpc-api/#snap_notify
+ * @see https://docs.metamask.io/snaps/reference/snaps-api/#snap_managestate
```
Comment (FrederikBolding) [inline packages/examples/packages/preinstalled/src/index.tsx:28]:
```suggestion
 * @see https://docs.metamask.io/snaps/reference/snaps-api/#snap_dialog
 * @see https://docs.metamask.io/snaps/reference/snaps-api/#snap_managestate
```

**Review by FrederikBolding — APPROVED** [review]

**Pull request merged**